### PR TITLE
chore: updated "HirFunction::unsafe_from_expr" to "HirFunction::unchecked_from_expr"

### LINKED ISSUE #812: Rename `HirFunction::unsafe_from_expr` to `HirFunction::from_expr_unchecked`
# Problem

The word `unsafe` here is being used to mean `unchecked`. Since `unsafe` has a particular meaning in Rust, we should be careful not to cause confusion when using it.

# Solution

Change `unsafe` word to `unchecked`

Thoughts welcome on bikeshedding, ie we could do `HirFunction::from_expr_unchecked` or `HirFunction::unchecked_from_expr` or something else; the core of this issue is to remove the `unsafe` keyword

# Alternatives considered

# Additional context

### PR DESCRIPTION
**Pull request by DreamerChaserHAH**
As per this issue https://github.com/noir-lang/noir/issues/812

Renamed

``HirFunction::unsafe_from_expr``
You can use the search function in your IDE to search for unsafe_from_expr, I cannot find it elsewhere only three places

to
``HirFunction::unchecked_from_expr``

# Related issue(s)

Resolves #812 

# Description

### Changes 1

``crates\noirc_frontend\src\hir\resolution\resolver.rs line number 299``
from
``HirFunction::unsafe_from_expr(expr_id)``
to
``HirFunction::unchecked_from_expr(expr_id)``

### Changes 2

``crates\noirc_frontend\src\hir\type_check\mod.rs line number 137``
from
``let func = HirFunction::unsafe_from_expr(expr_id);``
to
``let func = HirFunction::unchecked_from_expr(expr_id);``

### Changes 3

Added a comment in ``crates\noirc_frontend\src\hir_def\function.rs line number: 21 ``

### Changes 4

``crates\noirc_frontend\src\hir_def\function.rs line number 23/24``
from
``pub const fn unsafe_from_expr(expr_id: ExprId) -> HirFunction {``
to
``pub const fn unchecked_from_expr(expr_id: ExprId) -> HirFunction {``

Basically just changing every unsafe_from_expr to unchecked_from_expr, nothing else has been changed

## Test additions / changes

# Checklist

- [ ] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

File: crates/noirc_frontend/src/hir_def/function.rs [hunk 1]
```diff
@@ -18,9 +18,10 @@ impl HirFunction {
         HirFunction(ExprId::empty_block_id())
     }
 
+    // <Note> Originally named unsafe_from_expr changed to unchecked_from_expr as per this issue https://github.com/noir-lang/noir/issues/812 </Note>
```
Comment (kevaundray) [inline crates/noirc_frontend/src/hir_def/function.rs:21]: We can get rid of this comment because this change will be reflected in the CHANGELOG.md

**Review by kevaundray — CHANGES_REQUESTED** [review]
Thanks for resolving this, looks good overall. Left a comment.

File: crates/noirc_frontend/src/hir_def/function.rs (hunk 1 above)
Comment (DreamerChaserHAH) [reply crates/noirc_frontend/src/hir_def/function.rs]: Removed as per your request :D

**Review by kevaundray — APPROVED** [review]
LGTM
